File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        res = 0
        if x < 0:
            res = int(str(x)[1:][::-1]) * -1
        else:
            res = int(str(x)[::-1])
        
        if res > 2 ** 31 - 1 or res < -2 ** 31:
            return 0
        
        return res
        

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int num = Math.abs(x);
        long rev = 0;
        int obt;
        while (num > 0) {
            obt = num % 10;
            num = num / 10;
            rev = (rev * 10) + obt;
        }
        if (rev >= Integer.MIN_VALUE && rev <= Integer.MAX_VALUE) {
            return x < 0 ? (int)(-rev) : (int)rev;
        } else {
            return 0;
        }
    }
}